Sources/Logging/Log.swift

```diff
@@ -8,136 +8,137 @@
 
 import Foundation
 
-public final class Log: Logger {
+public enum Log {
 
-    public enum Error: Swift.Error {
+    // MARK: - Default Logger
+
+    public enum DefaultLoggerError: Swift.Error {
         case duplicateDestination(LogDestination.ID)
         case inexistentDestination(LogDestination.ID)
+        case duplicateModule(String)
+        case inexistentModule(String)
     }
 
-    /// Queue object used to specify `DispatchQueue`'s used in log destinations, ensuring they are serial queues of the
-    /// specified QoS, targeting an optional queue.
-    public final class Queue {
-        public let dispatchQueue: DispatchQueue
+    public final class DefaultLogger<Module: LogModule>: Logger {
 
-        public init(label: String, qos: DispatchQoS = .background, target: DispatchQueue? = nil) {
-            dispatchQueue = DispatchQueue(label: label, qos: qos, target: target)
-        }
-    }
+        public private(set) var destinations = Atomic<[LogDestination]>([])
+        public private(set) var modules = Atomic<[Module : Log.Level]>([:])
 
-    public private(set) var destinations = Atomic<[LogDestination]>([])
-    public var errorClosure: ((LogDestination, Item, Swift.Error) -> ())? = { destination, item, error in
-        print("💥[Alicerce.Log]: Failed to log item \(item) to destination '\(destination.id)' with error: \(error)")
-    }
+        public var errorClosure: ((LogDestination, Swift.Error) -> ())? = { destination, error in
+            print("💥[Alicerce.Log]: Failed to perform operation in destination '\(destination.id)' with error: \(error)")
+        }
 
-    public init() {}
+        public init() {}
 
-    // MARK: - Destination Management
+        // MARK: - Destination Management
 
-    public func register(_ destination: LogDestination) throws {
+        public func registerDestination(_ destination: LogDestination) throws {
```

Comment:
😢

Reply:
Yeah, I know 😢 

However I felt the final API wouldn't perhaps be very clear if we used two overrides for `register(_:)` and `unregister(_:)`

Do you think it's worth it? (i.e. makes the API as clear as it is now, from the point of calling)

Reply:
With the addition of the modules, this change makes sense indeed. Nevermind!